src/js/player.js

```diff
@@ -2416,6 +2417,12 @@ class Player extends Component {
       if (seconds < 0) {
         seconds = 0;
       }
+      if (!this.isReady_ || this.changingSrc_ || !this.tech_ || !this.tech_.isReady_ && seconds > 0) {
+        this.cache_.initTime = seconds;
+        this.off('canplay', this.applyInitTime_);
+        this.one('canplay', this.applyInitTime_);
```

Comment:
My one concern with canplay event is its reliability, since we want to seek to the given time as early as possible to ensure that no playback happens before we can seek. It's possible that this isn't really an issue anymore in modern browsers but I think it used to be an issue, particularly on iOS, where canplay only happens on play, so, we may end up seeing some content from the beginning of the video before we end up seeking. I'll have to test it out and see how it is.

Reply:
In my tests the `canplay` event was fired even when the play action was not triggered, this happens on all the browsers, but I forgot to test it on mobile especially on iOS

Reply:
yeah, it's probably much improved than the last time I was involved this type of seeking: https://github.com/brightcove/videojs-urlparams/blob/master/videojs.urlparams.js#L80-L126